src/utils/createBroadcastChannel.ts

```diff
@@ -24,7 +24,7 @@ export const createBroadcastChannel = (event: MessageEvent) => {
       payload?: Record<string, any> | string
     }) {
       if (port) {
-        port.postMessage(JSON.stringify(message))
+        port.postMessage(message)
```

Comment:
Okay, so when we sent a message from the worker, it's stringified and gets parsed in `handleRequestWith`, but when we communicate the response back to the worker, it's neither stringified nor parsed. I like that, wonder if the tests pass with this setup (will check).

Reply:
I initially left it untouched as it seems to be not necessary to change this to support binary response types. However I tried to remove the serialization inside of `handleRequestWith()` now but I couldn't get it to work without it.

Reply:
For instance, that other serialization is used to convert `request.headers` to string, and then construct a `Headers` instance out of it during deserialization on the client.

Reply:
Ah, I see. I think it absolute makes sense to have a custom serialization for `Headers` as they will not be handled by the structural cloning algorithm that comes with `postMessage()`.